app/lib/dns-server.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  CreateHostedZoneCommand,
+  ChangeResourceRecordSetsCommand,
+  GetChangeCommand,
+  ListResourceRecordSetsCommand,
+  ResourceRecordSet,
+} from "@aws-sdk/client-route-53";
+import { route53Client } from "./route53Client";
+
+export const createHostedZone = async (domain: string) => {
+  try {
+    const command = new CreateHostedZoneCommand({
+      Name: domain,
+      CallerReference: new Date().toString(),
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.HostedZone?.Id) {
+      throw new Error("Could create hosted zone upon request");
+    }
+
+    return response.HostedZone?.Id.replace("/hostedzone/", "");
+  } catch (error) {
+    throw new Error(`Error while creating hosted zone, ${error}`);
+  }
+};
+
+export const resetHostedZone = async (hostedZoneId: string) => {
+  try {
+    const resourceRecordSets = await listRecords(hostedZoneId);
+
+    resourceRecordSets.forEach(async (resourceRecordSet) => {
+      if (resourceRecordSet.Type === RecordType.A) {
+        await deleteRecord(
+          RecordType.A,
+          resourceRecordSet.Name!,
+          resourceRecordSet.ResourceRecords![0].Value!,
+          hostedZoneId
+        );
+      } else if (resourceRecordSet.Type === RecordType.AAAA) {
+        await deleteRecord(
+          RecordType.AAAA,
+          resourceRecordSet.Name!,
+          resourceRecordSet.ResourceRecords![0].Value!,
+          hostedZoneId
+        );
+      } else if (resourceRecordSet.Type === RecordType.CNAME) {
+        await deleteRecord(
+          RecordType.CNAME,
+          resourceRecordSet.Name!,
+          resourceRecordSet.ResourceRecords![0].Value!,
+          hostedZoneId
+        );
+      } else if (resourceRecordSet.Type === RecordType.TXT) {
+        await deleteRecord(
+          RecordType.TXT,
+          resourceRecordSet.Name!,
+          resourceRecordSet.ResourceRecords![0].Value!,
+          hostedZoneId
+        );
+      }
+    });
+  } catch (error) {
+    throw new Error(`Error while deleting hosted zone: ${error}`);
+  }
+};
+
+export const createRecord = async (
+  type: RecordType,
+  name: string,
```

Comment:
Let's add a function to validate the `name` callers can pass.  Route53 allows [this set](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/DomainNameFormat.html#domain-name-format-hosted-zones), but we can limit it a bit further.  I'd suggest:

```ts
// Only a-z, 0-9, and a hyphen - are valid in names
const isValidName = (name) => /^[a-z0-9-]+$/.test(name)
```

If a name other than this is given, we should throw.

Reply:
I would like to combine all validation into one function and common error message for both `name` and `value`. Let me know if this is ok.

Reply:
I wouldn't, they aren't the same.  Functions should do one thing when possible.  Easier to write tests that way, too.